### PR TITLE
Colorbar refactor

### PR DESCRIPTION
**Pull request by etpinard**
In brief, `colorbar` logic is now invoked in a very similar way to the other "margin-pusher" components (e.g. legend, updatemenus, etc.) - which should help us add colorbar features in the (near) feature.

See the commit description of https://github.com/plotly/plotly.js/commit/f7c5a8eeb556b2010883632c5267bce76c015abb for more details.

cc @plotly/plotly_js 

---

Going forward, I'm thinking of perhaps merging `components/colorbar/` into `components/colorscale/`. I can't think of a good reason for keeping them distinct. Some reasons: (1) `Colorscale` requires the `Colorbar` defaults, (2) most (if not all) trace modules manually extend their colorscale attributes with the `colorbar` attributes, (3) `Colorscale` doesn't have a `draw` method. Let me know what you think.

### CODE REVIEW

File: src/plots/plots.js [hunk 1]
```diff
@@ -54,18 +54,18 @@ plots.hasSimpleAPICommandBindings = commandModule.hasSimpleAPICommandBindings;
...
     return new Promise(function(resolve) {
         setTimeout(function() {
             Registry.getComponentMethod('annotations', 'draw')(gd);
             Registry.getComponentMethod('legend', 'draw')(gd);
-
-            (gd.calcdata || []).forEach(function(d) {
-                if(d[0] && d[0].t && d[0].t.cb) d[0].t.cb();
-            });
-
+            Registry.getComponentMethod('colorbar', 'draw')(gd);
```
Comment (etpinard) [inline src/plots/plots.js]: This commit essentially reverts https://github.com/plotly/plotly.js/pull/3786/commits/84c97088186336ced5a281ebdd0065c3dae96f14

No need to update the baselines!

**Comment by archmoj** [comment]
Looks very good to me.
:dancer:

**Comment by antoinerg** [comment]
This PR :hocho: 116 lines of code without changing any baselines. Great job @etpinard 

> Going forward, I'm thinking of perhaps merging components/colorbar/ into components/colorscale/. I can't think of a good reason for keeping them distinct. Some reasons: (1) Colorscale requires the Colorbar defaults, (2) most (if not all) trace modules manually extend their colorscale attributes with the colorbar attributes, (3) Colorscale doesn't have a draw method. Let me know what you think.

Sounds like a good idea indeed.

:dancer:

**Comment by etpinard** [comment]
THanks for the reviews guys. I'm holding off merging this thing. I'll thinking of potential releasing a 1.47.4 tomorrow or Thursday before merging things for 1.48.0

**Comment by etpinard** [comment]
OK to merge @plotly/plotly_js ?

**Comment by archmoj** [comment]
Nicely done.

:dancer: :dancer:
:dancer:

**Pull request merged**